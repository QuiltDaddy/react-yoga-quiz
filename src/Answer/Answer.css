.pose {
  height: 225px;
  max-width: 270px;
}

.link {
  display:inline-block;
  margin: auto;
  width: 200px;
  background-color: gainsboro;
  text-decoration: none;
  font-family: 'Oswald', sans-serif;
  color: #24A159;
  border-radius: 8px;
  box-shadow: 0px;
}

.link:focus {
  outline: none;
}

.link:hover {
  background-color: #ff476f;
  -webkit-transition-duration: 0.8s; /* Safari */
  transition-duration: 0.8s;
  color: white;
  cursor: pointer;
}
