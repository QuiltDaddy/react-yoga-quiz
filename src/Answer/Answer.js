import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import { Link } from 'react-router-dom';
import theQuiz from '../theQuiz';
import './Answer.css';

class Answer extends Component {
  componentDidMount(){
    if (!this.props.testing) {
      ReactDOM.findDOMNode(this.refs['link']).focus();
    }
  }

  render() {
    const question = this.props.question > 0 ? this.props.question : 1
    const {correct, nextQuestion} = this.props;
    
    return (
      <>
        <img 
          className='pose'
          src={theQuiz[question].pose}
          alt={theQuiz[question].alt} />
        <br/>
        {(correct === 1) ? (
          <h3>
            Good Job! The correct answer is: {theQuiz[question].answers[theQuiz[question].correctAnswer]}.  You choose right!
          </h3>
        ):(
          <h3>
            Sorry, the correct answer is: {theQuiz[question].answers[theQuiz[question].correctAnswer]}.
          </h3>
        )}
        {(question !== 10) ? (
          <Link
            to={`/question/${question + 1}`}
            className='link'
            ref='link'
            onClick={nextQuestion}>
            Next Question
          </Link>
        ):(
          <Link
            to={'/results'}
            className='link'
            ref='link'>
            View Results
          </Link>
        )}
      </>
    );
  }
}

export default Answer;
