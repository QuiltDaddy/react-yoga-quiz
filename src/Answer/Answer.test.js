import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Answer from './Answer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <BrowserRouter>
      <Answer
        question={1}
        correct={1} />
    </BrowserRouter>
  , div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
const tree = renderer
  .create(
    <BrowserRouter>
      <Answer 
        question={1}
        correct={1}
        testing={true} />
    </BrowserRouter>)
  .toJSON();
expect(tree).toMatchSnapshot();  
});
