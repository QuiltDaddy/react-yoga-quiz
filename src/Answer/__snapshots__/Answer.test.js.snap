// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
Array [
  <img
    alt="People standing with feet hips-width apart, a slight bend in their knees, hands by their sides and palms facing forward."
    className="pose"
    src="pose1.png"
  />,
  <br />,
  <h3>
    Good Job! The correct answer is: 
    Mountain Pose
    .  You choose right!
  </h3>,
  <a
    className="link"
    href="/question/2"
    onClick={[Function]}
  >
    Next Question
  </a>,
]
`;
